specs/agents/logging.md

```diff
@@ -31,11 +30,115 @@ agents can treat that as a synonym for `error` or `fatal`.
 
 The `off` level is a switch to completely turn off logging.
 
-### Backwards compatibility
+## Backwards compatibility
 
 Most agents have already implemented `log_level`,
 accepting a different set of levels.
 Those agents should still accept their "native" log levels to preserve backwards compatibility.
 However, in central config,
 there will only be a dropdown with the levels that are consistent across agents.
 Also, the documentation should not mention the old log levels going forward.
+
+## Logging Preamble
+
+All log messages described in this section MUST be printed using the `info` logging level
+unless a different level is explicitly mentioned.
+
+The agent logging preamble consists of 3 blocks:
+
+* **Agent**: This block is mandatory and contains basic version and build date information.
+* **Environment**: This block is optional but for supportability reasons it should be provided.
+* **Configuration**: This block mandatory and contains a minimum set of relevant configuration values.
+
+### Agent
+
+On startup, all APM agents MUST log basic information regarding their technology (language, runtime),
+version information, and build date in a format chosen by the respective agent.
+
+This SHOULD be the very first log message that is created by an agent.
+
+Example:
+
+```text
+Elastic APM .NET Agent, version: 1.19.1-preview, build date: 2022-10-27 10:55:42 UTC
+```
+
+Agents SHOULD also report when they are running in a **non-production** scenario (e.g. a debug
+or pre-release build) using the `warning` logging level.
+
+Example:
+
+```text
+This a pre-release version and not intended for use in production environments!
+```
+
+### Environment
+
+Additionally, agents SHOULD report information about their environment (e.g. host, process, runtime).
+
+| Item | Description | Example |
+| - | - | - |
+| Process ID | The Process ID in decimal format. | `83606` |
+| Process Name | The executable image name or the full path to it.  | `w3wp.exe`, `/usr/local/share/dotnet/dotnet` |
+| Command Line | The full command line used to launch this process as available to the runtime. | `/Users/acme/some_app/bin/Debug/net7.0/some_app.dll foo=bar` |
```

Comment:
Note that it was suggested in https://github.com/elastic/apm-agent-nodejs/issues/1916 that process argv could be considered personally identifiable information (PII).  I'm not sure whether that means we should exclude it here.

Reply:
I was not aware of that tbh.
[OpenTelemetry](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/process.md) e.g. seems to have a different opinion on that, though.
However, it is a SHOULD requirement. So if an agent decides against reporting this information with good reason, it's fine.

Reply:
@gregkalapos do you have an opinion on that?
Should we leave this in (as optional/SHOULD) or remove it altogether?

Reply:
I think keeping it as SHOULD is fine. This is agent log, so users still have control over it.